CMakeLists.txt

```diff
@@ -55,19 +55,43 @@ endif()
 
 
 file(GLOB_RECURSE infoware_source_files LIST_DIRECTORIES false src/*.cpp)
-add_library(infoware ${infoware_source_files})
+
+set(PRIVATE_HEADERS
+	include/infoware/detail/cpuid.hpp
+	include/infoware/detail/memory.hpp
+	include/infoware/detail/pci.hpp
+	include/infoware/detail/scope.hpp
+	include/infoware/detail/sysctl.hpp
+	include/infoware/detail/winstring.hpp)
+
+set(PUBLIC_HEADERS
```

Comment:
btw, this works for me when the variable is called `infoware_public_headers`

Reply:
Yes you could make a "glob" pattern to automatically have all the headers within `include` but it's a common good practice in CMake to have all the files listed explicitly instead (I encouraged strongly to see the [Daniel  Pfeifer's talk ](https://www.youtube.com/watch?v=bsXLMQ6WgIk) ).

  At first it could seem like unnecessary work but it's more effective in the end when the project grows.  His argument is that when you add a header or a source file, you'll have to regenerate the makefiles using CMake, but when you're using GLOB, the compilation will have to start from beginning, whereas if you add just a source file. CMake understands it better and it will just trigger the compilation for the new compilation unit. 

Another advantage of listing explicitly the source files in your case would be to only include the relevant files depending on the platform/options and thus avoiding some warnings like  `amounts_windows.cpp.o has no symbols` and avoiding putting `#ifdef/#endif ` in those files.

Hope I make myself clear

Reply:
Adding `CONFIGURE_DEPENDS` as an argument to `GLOB` removes the need to manually regenerate source files.

However, it requires a version of CMake from this side of the decade.

Reply:
Oh I didn't know that, thansks! Let me know if you'd prefer using this then

Reply:
hmm, played around with this and arrived at the following for this block

```cmake
file(GLOB_RECURSE infoware_source_files    LIST_DIRECTORIES false CONFIGURE_DEPENDS src/*.cpp)
file(GLOB_RECURSE infoware_private_headers LIST_DIRECTORIES false CONFIGURE_DEPENDS include/infoware/detail/*.hpp)

file(GLOB_RECURSE infoware_public_headers LIST_DIRECTORIES false CONFIGURE_DEPENDS include/infoware/*.hpp)
list(REMOVE_ITEM infoware_public_headers ${infoware_private_headers})

add_library(infoware ${infoware_source_files} ${infoware_public_headers} ${infoware_private_headers})
add_library(iware::infoware ALIAS infoware)
```

Which leaves us with maybe a way to put the installed headers under `include/infoware/`?